network.go

```diff
@@ -52,6 +52,8 @@ type NetworkController interface {
 	// Create a new network. The options parameter carry driver specific options.
 	// Labels support will be added in the near future.
 	NewNetwork(d *NetworkDriver, name string, options interface{}) (Network, error)
+	// Register an external driver.
+	RegisterDriver(name string, driver driverapi.Driver)
```

Comment:
Thats correct, there is no dependency on how plugins work and also, we don't want to expose such low level driver details through the APIs.  The Plugin discussions that are happening currently is narrowing down on the remote driver mechanism. Hence having an explicit RegisterDriver isn't a requirement (unless the plugin scope expands). Thats the reason for taking this cautious approach. @dave-tucker summarized it well in his comment. I think we will have a more solid answer to this question soon.